dspace-api/src/main/java/org/dspace/discovery/FullTextContentStreams.java

```diff
@@ -65,27 +71,41 @@ protected void init(Item parentItem) {
         }
     }
 
-    private void buildFullTextList(Item parentItem) {
+    private void buildFullTextList(Item parentItem) throws SQLException {
         // now get full text of any bitstreams in the TEXT bundle
         // trundle through the bundles
         List<Bundle> myBundles = parentItem.getBundles();
 
+        String logPrefix;
         for (Bundle myBundle : emptyIfNull(myBundles)) {
             if (StringUtils.equals(FULLTEXT_BUNDLE, myBundle.getName())) {
                 // a-ha! grab the text out of the bitstreams
                 List<Bitstream> bitstreams = myBundle.getBitstreams();
-
                 for (Bitstream fulltextBitstream : emptyIfNull(bitstreams)) {
-                    fullTextStreams.add(new FullTextBitstream(sourceInfo, fulltextBitstream));
-
-                    log.debug("Added BitStream: "
+                    if (isIndexable(fulltextBitstream,myBundle)) {
+                        fullTextStreams.add(new FullTextBitstream(sourceInfo, fulltextBitstream));
+                        logPrefix = "Added BitStream";
+                    } else {
+                        logPrefix = "Skipped Restricted BitStream";
+                    }
+                    log.debug(logPrefix+": "
                             + fulltextBitstream.getStoreNumber() + " "
                             + fulltextBitstream.getSequenceID() + " "
                             + fulltextBitstream.getName());
                 }
             }
         }
     }
+    
+    private boolean isIndexable(Bitstream fulltextBitstream,Bundle myBundle) throws SQLException {
+        for (ResourcePolicy rp:bundleService.getBitstreamPolicies(context, myBundle)) {
+            if (rp.getdSpaceObject().getID() == fulltextBitstream.getID() && rp.getGroup().getName().toLowerCase().equals("anonymous") 
+                    && resourcePolicyService.isDateValid(rp)) {
```

Comment:
Okay. Our use case at A&M was more straightforward. We simply didn't want the texts indexed if they weren't also publicly available. And we do update our SOLR index nightly.

The short term advice in the JIRA issue for those concerned was to just turn snippet indexing off altogether while a solution was researched. Would adding an on/off configuration parameter to this implementation make it more broadly useful?

For the second bullet point, could this be solved by allowing the policy names to be excluded/allowed to also be set through configuration?